editor/editor_properties_array_dict.cpp

```diff
@@ -240,6 +234,48 @@ void EditorPropertyArray::_object_id_selected(const StringName &p_property, Obje
 	emit_signal(SNAME("object_id_selected"), p_property, p_id);
 }
 
+void EditorPropertyArray::create_new_property_slot() {
+	int idx = slots.size();
+	HBoxContainer *hbox = memnew(HBoxContainer);
+
+	Button *reorder_button = memnew(Button);
+	reorder_button->set_icon(get_editor_theme_icon(SNAME("TripleBar")));
+	reorder_button->set_default_cursor_shape(Control::CURSOR_MOVE);
+	reorder_button->set_disabled(is_read_only());
+	reorder_button->connect("gui_input", callable_mp(this, &EditorPropertyArray::_reorder_button_gui_input));
+	reorder_button->connect("button_up", callable_mp(this, &EditorPropertyArray::_reorder_button_up));
+	hbox->add_child(reorder_button);
+	EditorProperty *prop = memnew(EditorPropertyNil);
+	hbox->add_child(prop);
+
+	reorder_button->connect("button_down", callable_mp(this, &EditorPropertyArray::_reorder_button_down).bind(idx));
+
+	bool is_untyped_array = object->get_array().get_type() == Variant::ARRAY && subtype == Variant::NIL;
+
+	if (is_untyped_array) {
+		Button *edit_btn = memnew(Button);
+		edit_btn->set_icon(get_editor_theme_icon(SNAME("Edit")));
+		edit_btn->set_disabled(is_read_only());
+		edit_btn->connect("pressed", callable_mp(this, &EditorPropertyArray::_change_type).bind(edit_btn, idx));
+		hbox->add_child(edit_btn);
+	} else {
+		Button *remove_btn = memnew(Button);
+		remove_btn->set_icon(get_editor_theme_icon(SNAME("Remove")));
+		remove_btn->set_disabled(is_read_only());
+		remove_btn->connect("pressed", callable_mp(this, &EditorPropertyArray::_remove_pressed).bind(idx));
+		hbox->add_child(remove_btn);
+	}
+	property_vbox->add_child(hbox);
+
+	Slot slot = Slot();
+	slot.prop = prop;
+	slot.object = object;
+	slot.hbox = hbox;
```

Comment:
```suggestion
	slot.container= hbox;
```
I think a field name should be more explicit than "hbox". Other name could be `control` or `content`, or maybe you have some better idea.